tracing-subscriber/src/registry/mod.rs

```diff
@@ -291,43 +471,87 @@ where
     }
 }
 
-impl<'a, R> Iterator for Parents<'a, R>
-where
-    R: LookupSpan<'a>,
-{
-    type Item = SpanRef<'a, R>;
-    fn next(&mut self) -> Option<Self::Item> {
-        let id = self.next.take()?;
-        let span = self.registry.span(&id)?;
-        self.next = span.parent().map(|parent| parent.id());
-        Some(span)
-    }
-}
-
-// === impl FromRoot ===
-
-impl<'span, R> Iterator for FromRoot<'span, R>
-where
-    R: LookupSpan<'span>,
-{
-    type Item = SpanRef<'span, R>;
+#[cfg(test)]
+mod tests {
+    use crate::{
+        layer::{Context, Layer},
+        prelude::*,
+        registry::LookupSpan,
+    };
+    use std::sync::{Arc, Mutex};
+    use tracing::{span, Subscriber};
 
-    #[inline]
-    fn next(&mut self) -> Option<Self::Item> {
-        self.inner.next()
-    }
-
-    #[inline]
-    fn size_hint(&self) -> (usize, Option<usize>) {
-        self.inner.size_hint()
+    #[test]
+    fn spanref_scope_iteration_order() {
+        let last_entered_scope = Arc::new(Mutex::new(Vec::new()));
+        #[derive(Default)]
+        struct PrintingLayer {
+            last_entered_scope: Arc<Mutex<Vec<&'static str>>>,
+        }
+        impl<S> Layer<S> for PrintingLayer
+        where
+            S: Subscriber + for<'lookup> LookupSpan<'lookup>,
+        {
+            fn on_enter(&self, id: &span::Id, ctx: Context<'_, S>) {
+                let span = ctx.span(id).unwrap();
+                let scope = span.scope().map(|span| span.name()).collect::<Vec<_>>();
+                *self.last_entered_scope.lock().unwrap() = scope;
+            }
+        }
+        tracing::subscriber::with_default(
+            crate::registry().with(PrintingLayer {
+                last_entered_scope: last_entered_scope.clone(),
+            }),
+            || {
+                let _root = tracing::info_span!("root").entered();
+                assert_eq!(&*last_entered_scope.lock().unwrap(), &["root"]);
+                let _child = tracing::info_span!("child").entered();
+                assert_eq!(&*last_entered_scope.lock().unwrap(), &["child", "root"]);
+                let _leaf = tracing::info_span!("leaf").entered();
+                assert_eq!(
+                    &*last_entered_scope.lock().unwrap(),
+                    &["leaf", "child", "root"]
+                );
+            },
+        );
     }
-}
 
-impl<'span, R> std::fmt::Debug for FromRoot<'span, R>
-where
-    R: LookupSpan<'span>,
-{
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.pad("FromRoot { .. }")
+    #[test]
+    fn spanref_scope_fromroot_iteration_order() {
+        let last_entered_scope = Arc::new(Mutex::new(Vec::new()));
+        #[derive(Default)]
+        struct PrintingLayer {
+            last_entered_scope: Arc<Mutex<Vec<&'static str>>>,
+        }
+        impl<S> Layer<S> for PrintingLayer
```

Comment:
nit: again, whitespace here would be nice
```suggestion
        let last_entered_scope = Arc::new(Mutex::new(Vec::new()));
        
        #[derive(Default)]
        struct PrintingLayer {
            last_entered_scope: Arc<Mutex<Vec<&'static str>>>,
        }
        
        impl<S> Layer<S> for PrintingLayer
```